### ISSUE TITLE
[FABRIC] normal tips cannot be disabled and no mod menu integration

### ISSUE THREAD

**Issue by szunakabeats** [Darkhax-Minecraft/Tips#57]
with the [modmenu](https://www.curseforge.com/minecraft/mc-mods/modmenu) you can not open settings for tips in fabric

also putting "tips" in : "ignoredNamespaces" does not disable the default tips 

here is my config:
{
  "defaultCycleTime": 5200,
  "ignoredNamespaces": [tips],
  "ignoredTips": []
}

using Fabric 0.14.8 and Tips-Fabric-1.19-6.0.1

my custom tips work as intended tho :)

**Comment by szunakabeats** [comment]
oh yea and one quick question is this the right configuration for setting the cycleTime ?

`{
  "title": {
    "text": "Biomes",
    "bold": true,
    "underlined": true,
    "color": "dark_green"
  },
  "tip": {
    "cycleTime": 4600,
    "text": "You can find 12 new Biomes on the Eden Ring!"
  }
}`

**Comment by Darkhax** [comment]
Hello, the namespace in 1.18/1.19 is `tipsmod` instead of `tips`. This change was to help make things less confusing for those making custom tip packs.

**Comment by szunakabeats** [comment]
hello now it just says " No tips loaded. Please review your config options! " my custom tips aren't working anymore aswell :/

**Comment by Darkhax** [comment]
What is the ID of your custom tip? If it's in the tipsmod folder then it would have the tipsmod namespace and be removed as well. You should be using a different folder for custom tips. For example `assets/example/tips/my_tip.json` would be `example:my_tip`

**Comment by szunakabeats** [comment]
the tips mod does not create a folder in fabric 1.19... my custom tips are in:

"1.19 modpack\config\openloader\resources\restored_tips\assets\restored\tips"

**Comment by szunakabeats** [comment]
the weird thing is sometimes it loads the custom tips and cycles through 4-5 of them and then the "No Tips loaded" message comes again. as if the mod is loosing track of the openloader folder or something

here are my custom tips: https://imgur.com/FX4MaZf

**Comment by szunakabeats** [comment]
would be awesome to get some help here ..

**Comment by Darkhax** [comment]
I haven't had time to look more into your issue. Please try out the new 6.0.2 patch and see if that helps your issue.

**Comment by szunakabeats** [comment]
no problem ! the issue seems to be solved now :) thx !

**Comment by szunakabeats** [comment]
but a mod menu integration still would be nice

**Comment by ghost** [comment]
@szunakabeats Sorry for bothering you but would you ever consider sharing those tips files? Some of them could be useful to others!

Dreaming here, but a central repo for tips JSON files for different mods where people can contribute would be really cool.

**Comment by szunakabeats** [comment]
@Brittank88 well it is really easy to write your own ones and what i did is i downloaded a modpack for example "Crucial 2" and looked in it's files to see how they wrote their tips :) but if you want i can send you some examples !

**Comment by ghost** [comment]
That's kind of you! I would like to know how to write tips that can support depending on the presence of a mod in both Forge and Fabric. Here's my attempt at that (untested):

```json
{
	"conditions": [
		{
			"type": "minecraft:alternative"
			"terms": [
				{
					"type": "fabric:all_mods_loaded",
					"values": [ "antighost" ]
				},
				{
					"type": "forge:mod_loaded",
					"modid": "antighost"
				}
			]
		}
	],
	"tip": {
		"translate": "res_mechanica-tips-command.tip.antighost"
	}
}
```

My idea was more in reference to providing a library of standard tips for popular mods that help users in understanding them, and modpack creators in not having to manually write out a bunch of tips for certain mods.

**Comment by szunakabeats** [comment]
thats actually a crazy idea that you got ! unfortunately I'm not a professional in such technical matters so i can't help you with your attempt :/

**### here is one of my tips:**

{
  "title": {
    "text": "Controls",
    "bold": true,
	"underlined": true,
    "color": "red"
  },
  "tip": {
    "text": "In the Controls menu you can find the Controller Options to set up your favorite Controller"
  }
}

**### here is another one :**

{
  "title": {
    "text": "Tools",
    "bold": true,
	"underlined": true,
    "color": "#fc8dd4"
  },
  "tip": {
    "text": "The Amethyst Lamp doesn't light up the area, but decreases mob spawning in a radius of 8 blocks!"
  }
}

**Comment by szunakabeats** [comment]
thats a crazy idea ! Unfortunately I'm not a professional in such technical matters so I can't help you with your attempt :/

here are 2 of my tips:

https://we.tl/t-1LU9rxFUSv

**Comment by ghost** [comment]
Thanks for showing me them! I'm thinking it might be a good idea to take a leaf out of your book and treat the tip titles as categories. Good idea!

**Comment by Darkhax** [comment]
@Brittank88

> Dreaming here, but a central repo for tips JSON files for different mods where people can contribute would be really cool.

Anyone can make a Pull Request on GitHub to contribute new tips, and we've already had a dozen or so community members contribute some already.

> I would like to know how to write tips that can support depending on the presence of a mod in both Forge and Fabric.

Recipe conditions like this are not a vanilla feature, and Forge / Fabric could not agree to use the same JSON format. To get around this I created a loader neutral layer in the Bookshelf mod which you should already have as it's a dependency. Bookshelf conditions look like this. Stuff like this should work on nearly all json files in resource packs and data packs as well.

```json
{
  "bookshelf:load_conditions": [
    {
      "type": "bookshelf:mod_loaded",
      "mods": "botanypots"
    }
  ],
  
  // rest of JSON file here
}
```

**Comment by ghost** [comment]
Oh, that's wonderful! Thank you so much for clearing it all up.

**Comment by ghost** [comment]
> Anyone can make a Pull Request on GitHub to contribute new tips, and we've already had a dozen or so community members contribute some already.

I will say though, that it seems that this only applies to vanilla tips, right? I was more talking about mod-focused tips, though maybe they can be PR'd too and it's just that I'm not seeing them in the assets folder?

**Comment by Darkhax** [comment]
@Brittank88 Generally mods will implement the tips themselves, although I am not against adding some modded tips as well. The main thing is that I don't want to give players incorrect or outdated information.

**Comment by ghost** [comment]
Oh, absolutely.

Also, I am really struggling to get even one custom tip to load. Can you even reload them or the config whilst in-game?

I have it set up in a resourcepack, with the following paths:

`assets/tipsmod-command/tips/antighost.json`:
```json
{
	"bookshelf:load_conditions": [
		{
			"type": "bookshelf:mod_loaded",
			"mods": "antighost"
		}
	],
    "tip": {
        "translate": "tipsmod-command.tip.antighost"
    }
}
```

`assets/tipsmod-command/lang/en_us.json`:
```json
{
    "tipsmod-command.tip.antighost": "Use the /ghost command to reload ghost blocks in your radius."
}
```

**Comment by ghost** [comment]
Nevermind, I got it working!

**Comment by aaronhowser1** [comment]
> Nevermind, I got it working!

![image](https://user-images.githubusercontent.com/15698722/185405243-d398e875-ac06-4dad-b45b-2c8a722604b0.png)

**Comment by ghost** [comment]
Yo if you still want the tips file I ended up with I can go find it and send it to you on Discord ahaha

**Comment by Darkhax** [comment]
There was another bug in the mod preventing things from working as intended. This bug has been fixed on 1.18.2, and should be fixed on 1.19+ soon.
